CDFModule/Public/func_Get-ServiceConfigSettings.ps1

```diff
@@ -53,95 +58,156 @@ Function Get-ServiceConfigSettings {
+                    }
+                    else {
+                        $UpdateSettings[$appSettingKey] = [System.Net.NetworkCredential]::new("", $secret.SecretValue).Password
+                        Write-Verbose "Prepared KeyVault secret [$kvSecretName] for Setting [$appSettingKey]"
+                    }
+                }
+                else {
+                    $appSettingRef = "@Microsoft.KeyVault(VaultName=$($CdfConfig.Domain.ResourceNames.keyVaultName );SecretName=$kvSecretName)"
+                    $UpdateSettings[$appSettingKey] = $appSettingRef
+                    Write-Verbose "Prepared KeyVault secret reference for Setting [$kvSecretName] using app setting [$appSettingKey] KeyVault ref [$appSettingRef]"
+                }
             }
         }
     }
-    
-    # CDF Env details
-    $UpdateSettings["CDF_ENV_DEFINITION_ID"] = $CdfConfig.Application.Env.definitionId
-    $UpdateSettings["CDF_ENV_NAME_ID"] = $CdfConfig.Application.Env.nameId
-    $UpdateSettings["CDF_ENV_NAME"] = $CdfConfig.Application.Env.name
-    $UpdateSettings["CDF_ENV_SHORT_NAME"] = $CdfConfig.Application.Env.shortName
-    $UpdateSettings["CDF_ENV_DESCRIPTION"] = $CdfConfig.Application.Env.description
-    $UpdateSettings["CDF_ENV_PURPOSE"] = $CdfConfig.Application.Env.purpose
-    $UpdateSettings["CDF_ENV_QUALITY"] = $CdfConfig.Application.Env.quality
-    $UpdateSettings["CDF_ENV_REGION_CODE"] = $CdfConfig.Application.Env.regionCode
-    $UpdateSettings["CDF_ENV_REGION_NAME"] = $CdfConfig.Application.Env.regionName
 
-    # Service Identity
-    $UpdateSettings["CDF_SERVICE_NAME"] = $CdfConfig.Service.Config.serviceName
-    $UpdateSettings["CDF_SERVICE_TYPE"] = $CdfConfig.Service.Config.serviceType
-    $UpdateSettings["CDF_SERVICE_GROUP"] = $CdfConfig.Service.Config.serviceGroup
-    $UpdateSettings["CDF_SERVICE_TEMPLATE"] = $CdfConfig.Service.Config.serviceTemplate
-    $UpdateSettings["CDF_DOMAIN_NAME"] = $CdfConfig.Domain.Config.domainName       
-    
-    # Build information
-    $UpdateSettings["CDF_BUILD_COMMIT"] = $env:GITHUB_SHA ?? $env:BUILD_SOURCEVERSION ?? $(git -C $TemplateDir rev-parse --short HEAD)
-    $UpdateSettings["CDF_BUILD_RUN"] = $env:GITHUB_RUN_ID ?? $env:BUILD_BUILDNUMBER ?? "local"
-    $UpdateSettings["CDF_BUILD_BRANCH"] = $env:GITHUB_REF_NAME ?? $env:BUILD_SOURCEBRANCH ?? $(git -C $TemplateDir branch --show-current)
-    $UpdateSettings["CDF_BUILD_REPOSITORY"] = $env:GITHUB_REPOSITORY ?? $env:BUILD_REPOSITORY_NAME ?? $(Split-Path -Leaf (git -C $TemplateDir remote get-url origin))
-    $UpdateSettings["CDF_BUILD_PIPELINE"] = $env:GITHUB_WORKFLOW_REF ?? $env:BUILD_DEFINITIONNAME ?? "local"
-    $UpdateSettings["CDF_BUILD_BRANCH"] = $env:GITHUB_REF_NAME ?? $env:BUILD_SOURCEBRANCH ?? $(git -C $TemplateDir branch --show-current)
+    # Add default/override app settings if exists - override any generated app settings
+    if (Test-Path "$OutputPath/app.settings.json") {
```

Comment:
Are we still planning to use 2 different sources for configuration/settings? I was under the impression that we will introduce a new type of setting type (may be named Standard) in cdf.config and all those configuration/settings which needs to be used as it is can be specified under this new setting type.

Reply:
Is it already in place for use?
Don't we need to keep the appsettings for backwards compatibility?